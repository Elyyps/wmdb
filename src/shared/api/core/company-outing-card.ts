import { ILink } from "./link";

export interface ICompanyOutingCard {
  button?: ILink;
  content?: string;
  data?: string;
  icon?: string;
  image?: string;
  title?: string;
}
