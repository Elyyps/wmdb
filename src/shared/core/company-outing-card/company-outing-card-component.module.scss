.company-outing-card {
  display: block;
  border-radius: 4px;
  overflow: hidden;
  position: relative;
  background: $global-background;
  padding: 20px 24px;
  font-size: 14px;

  &__head {
    color: $gray-light;
    margin-bottom: 18px;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__date {
    color: $gray-light;
  }
  &__content {
    margin-bottom: 24px;
  }
  &__image {
    overflow: hidden;
    position: relative;
    height: 190px;
    margin: 0 -24px 24px;
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
}
