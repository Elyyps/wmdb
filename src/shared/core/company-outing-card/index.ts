export * from "./company-outing-card.component";
