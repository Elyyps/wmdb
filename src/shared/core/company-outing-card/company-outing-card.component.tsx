import * as React from "react";
import styles from "./company-outing-card-component.module.scss";
import { IconComponent } from "../icon";
import { ImageComponent } from "../image";
import Arrow from "@assets/icons/chevron-right.svg";
import { ILink } from "@app/api/core/link/link";
import { Button } from "@app/prep/modules-prep/core";

export interface ICompanyOutingCardComponentProps {
  button?: ILink;
  content?: string;
  data?: string;
  icon?: string;
  image?: string;
  title?: string;
}

const CompanyOutingCardComponent = ({
  title,
  image,
  button,
  data,
  content,
  icon
}: ICompanyOutingCardComponentProps) => (
  <div className={styles["company-outing-card"]}>
    <div className={styles["company-outing-card__head"]}>
      <div className={styles["company-outing-card__date"]}>{data}</div>
      {icon && <IconComponent icon={icon} size="24px" fillColor={"#969696"} />}
    </div>
    <div className={styles["company-outing-card__image"]}>
      <ImageComponent src={image} />
    </div>
    <h4 className={styles["company-outing-card__title"]}>{title}</h4>
    <div className={styles["company-outing-card__content"]}>{content}</div>
    {button && <Button icon={Arrow} title={button.title} href={button.url} position={"left"} />}
  </div>
);

export { CompanyOutingCardComponent };
