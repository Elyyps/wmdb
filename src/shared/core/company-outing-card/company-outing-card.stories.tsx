import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { CompanyOutingCardComponent } from "./company-outing-card.component";

storiesOf("CompanyOutingCard", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <CompanyOutingCardComponent />);
