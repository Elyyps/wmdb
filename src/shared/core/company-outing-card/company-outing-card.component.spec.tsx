import React from "react";
import { shallow } from "enzyme";
import { CompanyOutingCardComponent } from "./company-outing-card.component";

describe("[CompanyOutingCard]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<CompanyOutingCardComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
