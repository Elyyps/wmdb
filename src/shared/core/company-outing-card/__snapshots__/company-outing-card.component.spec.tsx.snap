// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CompanyOutingCard] should render component without crashing 1`] = `
<div
  className="company-outing-card"
>
  <div
    className="company-outing-card__head"
  >
    <div
      className="company-outing-card__date"
    />
  </div>
  <div
    className="company-outing-card__image"
  >
    <ImageComponent />
  </div>
  <h4
    className="company-outing-card__title"
  />
  <div
    className="company-outing-card__content"
  />
</div>
`;
